### DISCUSSION TITLE
Alias as abbreviation instead?

### DISCUSSION THREAD

**Discussion by khuongduybui** [nushell/nushell#2936] [Ideas]
I have fallen in love with fish shell's abbreviation approach for aliasing.
The main reason is simply illustrated like this:

Most of the time I would run `ls` command with `ls -la` options.
So I `alias` it to simply `l` => `l = ls -la`.

Now with `nu` 0.25.x approach to alias, in some cases where I want to run `ls -lad` instead, I can still invoke it with `l -d`.
However, in the super rare cases where I want to run `ls -l` (without `-a`), I cannot use `l` at all.

With fish shell abbreviation approach, when I type `l<space>` , the command is replaced with `ls -la<space>`, so I can conveniently edit the command with both additions and deletions.

Of course the example above does not make sense in practice, because it would be faster to simply type `ls -l` instead of `l<space><backspace><backspace>`.
However, please imagine a very very long alias where sometimes it makes sense to edit a tiny bit in the middle of the command, not at the end.

**Comment by fdncred** [comment]
Agreed. It would be nice to have such a feature. Currently we do not.

**Reply by cybardev** [reply]
Hi, I'm here from https://github.com/nushell/nushell/pull/16960#issuecomment-3673289917

This is exactly the abbr behaviour I was looking for. Has this been implemented yet, or is it what that PR is trying to achieve?

**Reply by Mrfiregem** [reply]
(Basic) Fish-style `abbr` is implemented in [0.113](https://www.nushell.sh/blog/2026-05-23-nushell_v0_113_0.html#added-fish-style-abbreviations) now :+1:
